### ISSUE TITLE
verbs build hangs in tests/charm++/communication_overhead

### ISSUE THREAD

**Issue by PhilMiller** [UIUC-PPL/charm#1473]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/1473*

---

```
==> old.2017_03_18__01_00/verbs-linux-x86_64.txt <==
Charm++ group communication with allocation timing enabled

 PE             MSG SIZE     PER MSG TIME(us)             BW(MB/s)         OVERHEAD(us)
slurmstepd: *** JOB 8367429 ON c457-401 CANCELLED AT 2017-03-18T08:55:38 DUE TO TIME LIMIT ***
make[2]: *** [test] Terminated
make[1]: *** [test] Terminated
make: make[3]: *** [test] Terminated*** [test] Terminated

fatal> error code 1 during remote> ./instead_test.sh charm/verbs-linux-x86_64/tmp make  test ++timeout 180 +isomalloc_sync
Returned from executing scripts/verbs-linux-x86_64/test on remote host
fatal> Test on remote host failed with fatal error (0)
Bad: Test on remote host failed with fatal error (0)

==> old.2017_03_19__01_00/verbs-linux-x86_64.txt <==
Charm++ group communication with allocation timing enabled

 PE             MSG SIZE     PER MSG TIME(us)             BW(MB/s)         OVERHEAD(us)
slurmstepd: *** JOB 8368997 ON c457-703 CANCELLED AT 2017-03-19T15:39:30 DUE TO TIME LIMIT ***
make[2]: make[1]: *** [test] Terminated*** [test] Terminated

make: *** [test] Terminatedmake[3]: 
*** [test] Terminated
fatal> error code 1 during remote> ./instead_test.sh charm/verbs-linux-x86_64/tmp make  test ++timeout 180 +isomalloc_sync
Returned from executing scripts/verbs-linux-x86_64/test on remote host
fatal> Test on remote host failed with fatal error (0)
Bad: Test on remote host failed with fatal error (0)

==> old.2017_03_20__01_00/verbs-linux-x86_64.txt <==
Charm++ 1D array communication with allocation timing enabled

 PE             MSG SIZE     PER MSG TIME(us)             BW(MB/s)         OVERHEAD(us)
slurmstepd: *** JOB 8370620 ON c461-402 CANCELLED AT 2017-03-20T18:12:32 DUE TO TIME LIMIT ***
make[2]: make[1]: *** [test] Terminated*** [test] Terminated

make: *** [test] Terminated
make[3]: *** [test] Terminated
fatal> error code 1 during remote> ./instead_test.sh charm/verbs-linux-x86_64/tmp make  test ++timeout 180 +isomalloc_sync
Returned from executing scripts/verbs-linux-x86_64/test on remote host
fatal> Test on remote host failed with fatal error (0)
Bad: Test on remote host failed with fatal error (0)
```

**Comment by PhilMiller** [comment]
*Original date: 2017-03-21 17:28:43*

---
Tried the same test back on 6.7.1. It also hangs, but somewhat later - in the 1D array cases, instead of the group cases.

Might there be a memory leak at issue here, possibly screwing with the registration pool?

**Comment by PhilMiller** [comment]
*Original date: 2017-03-21 17:52:51*

---
Observed what seems like a memory leak in operationFinished, in that the message it receives is sometimes not deleted. Fixing that doesn't make the hang go away, though.

**Comment by PhilMiller** [comment]
*Original date: 2017-03-22 16:29:09*

---
Test is indeed broken, disablement commit pushed for now.

**Comment by stwhite91** [comment]
*Original date: 2017-04-11 18:50:34*

---
This has been worked around for now.

**Comment by ericjbohm** [comment]
*Original date: 2018-03-14 19:02:30*

---
Current issue is not a hang, it crashes.


 PE             MSG SIZE     PER MSG TIME(us)             BW(MB/s)         OVERHEAD(us)
[1]                   16                3.634                4.199                0.496
Expected message of size 16, got message of size 32
------------- Processor 0 Exiting: Called CmiAbort ------------
Reason: Out of order message
[0] Stack Traceback:
  [0:0] CmiAbortHelper+0x65  [0x5a56a5]
  [0:1]   [0x5a56fb]
  [0:2] _ZN26CkIndex_CommunicationGroup37_call_operationFinished_SimpleMessageEPvS0_+0x42  [0x49cf32]
  [0:3] CkDeliverMessageFree+0x3a  [0x4afa2a]
  [0:4] _Z15_processHandlerPvP11CkCoreState+0x369  [0x4b7f39]
  [0:5] CsdScheduleForever+0x70  [0x5af4b0]
  [0:6] CsdScheduler+0x2d  [0x5af7fd]
  [0:7]   [0x5abb22]
  [0:8] ConverseInit+0x307  [0x5ad2d7]
  [0:9] main+0x27  [0x495027]
  [0:10] __libc_start_main+0xfd  [0x2b96c185ed1d]
  [0:11]   [0x4955ad]
Fatal error on PE 0> Out of order message

**Comment by ericjbohm** [comment]
Replicated on Converse/Charm++ Commit ID: v6.9.0-368-g65709c8 in benchmarks/communication_overhead.

The out of order complaint is issued by the application, currently studying it to determine if the bug is actually in the application failing to handle non-determinism rather than the machine layer.

**Comment by evan-charmworks** [comment]
Fixed by #2302
